src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -239,6 +245,16 @@ public BasePanel(JabRefFrame frame, BibDatabaseContext bibDatabaseContext) {
         }
 
         this.getDatabase().registerListener(new UpdateTimestampListener(Globals.prefs));
+
+        this.preview = new PreviewPanel(this, getBibDatabaseContext());
+        DefaultTaskExecutor.runInJavaFXThread(() -> frame().getGlobalSearchBar().getSearchQueryHighlightObservable().addSearchListener(preview));
+        this.previewContainer = setupPreviewContainer(preview);
+    }
+
+    private JFXPanel setupPreviewContainer(PreviewPanel preview) {
+        JFXPanel container = OS.LINUX ? new CustomJFXPanel() : new JFXPanel();
```

Comment:
I don't like the fact this chunk of three lines is duplicated all over the place. Maybe add a new factory class that provides a method to build a `JFXPanel` but hides the OS aspect.

Reply:
Good remark! Fixed.